openedx/core/lib/api/view_utils.py

```diff
@@ -104,16 +104,8 @@ def _wrapper(self, request, *args, **kwargs):
                         depth=depth,
                         check_if_enrolled=True,
                     )
-                except Http404:
-                    # any_unfulfilled_milestones called a second time since has_access returns a bool
-                    if check_for_milestones and any_unfulfilled_milestones(course_id, request.user.id):
-                        message = {
-                            "developer_message": "Cannot access content with unfulfilled "
-                                                 "pre-requisites or unpassed entrance exam."
-                        }
-                        return response.Response(data=message, status=status.HTTP_204_NO_CONTENT)
-                    else:
-                        raise
+                except Http404 as error:
+                    return response.Response(data=error.message, status=status.HTTP_204_NO_CONTENT)
```

Comment:
Note: Functionally this is different. In this new code we would be returning all errors at 204 while in the old code we only returned a milestone error as a 204. I believe we need to figure out how we want to handle errors. @aleffert FYI